bifrost/renderer/wrapped/onRenderClient.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { PropsWithChildren } from "react";
+import {
+  AugmentMe,
+  PageContextProxyClient,
+  PageContextProxyClientHydration,
+  PageContextProxyClientNavigation,
+  PageContextProxyClientRestorationVisit,
+} from "../../types/internal";
+import { PageShell } from "../../lib/PageShell";
+import { getElementAttributes } from "../../lib/getElementAttributes";
+import { renderReact } from "../../lib/renderReact";
+import { Turbolinks } from "../../lib/turbolinks";
+import { copyElementAttributes } from "../../lib/turbolinks/util";
+
+const PassThruLayout: React.ComponentType<PropsWithChildren> = ({
+  children,
+}) => <>{children}</>;
+
+/* 
+We get here in 3 ways:
+1. Hydration from initial SSR
+2. Client navigation
+3. Back button restoration
+
+Each path needs to generate pageContext differently
+*/
+
+export async function wrappedOnRenderClient(
+  pageContext: PageContextProxyClient
+) {
+  if ("snapshot" in pageContext) {
+    onRenderClientRestorationVisit(pageContext);
+  } else if (pageContext.isHydration) {
+    onRenderClientHydration(pageContext);
+  } else {
+    onRenderClientNavigation(pageContext);
+  }
+}
+
+async function onRenderClientHydration(
+  pageContext: PageContextProxyClientHydration
+) {
+  const { layoutProps, layout } = pageContext;
+
+  await pageContext.config.onClientInit?.();
+  const bodyEl = document.getElementById("proxied-body")!;
+  // During hydration of initial ssr, body is in dom, not page props (to avoid double-send)
+  render({
+    html: bodyEl.innerHTML,
+    layout,
+    layoutProps,
+    bodyEl,
+    pageContext,
+  });
+}
+
+async function onRenderClientNavigation(
+  pageContext: PageContextProxyClientNavigation
+) {
+  const resp = await fetch(pageContext.urlOriginal, {
```

Comment:
`onRenderClientNavigation` is the only real change. everything else is refactor / moving code around.

This is significant though. We do not rely on `index.pageContext.json` / onBeforeRender anymore. Instead, we hit the url directly, with the proper proxy header.